shared/elements/gaia-header/dist/gaia-header.js

```diff
@@ -1127,8 +1127,9 @@ module.exports = component.register('gaia-header', {
   sumButtonWidths: function(buttons) {
     var defaultWidth = 50;
     var sum = buttons.reduce((prev, button) => {
+      var isEmpty = !button.children.length && !button.textContent;
       var isStandardButton = button === this.els.actionButton;
-      var width = isStandardButton ? defaultWidth : button.clientWidth;
+      var width = isEmpty || isStandardButton ? defaultWidth : button.clientWidth;
```

Comment:
Ah sorry I didn't read the code properly, I think some braces are in order :)

```js
var width = (isEmpty || isStandardButton) ? defaultWidth : button.clientWidth;
```

Besides, @julienw is right, if an empty button is hidden with CSS (which they frequently are) it's `clientWidth` will be `0` not `defaultWidth` (`50`). If we could enforce header buttons to be hidden with a `hidden` attribute they we could detect hidden buttons without using `clientWidth`. Unfortunately I don't think this realistic.